docs/proposal-optional-output-extension.md

```diff
@@ -41,16 +41,33 @@ There are currently three problems with expiry (as of cylc-flow 8.1.4):
    executed and that might not happen if expiry or submission are optional (or if the pre-requisites
    for `a` are not met).
 
-3. A task is incomplete if the completion condition is not satisfied and:
+3. A task is complete if the completion condition is satisfied.
```

Comment:
```suggestion
3. Outputs are complete if the completion condition is satisfied.
```

Reply:
@hjoliver is this correction sufficient for you?

Reply:
Not by itself - it has to be carried throughout the document wherever "complete tasks" and "incomplete tasks" are referred to.

I take it you are agreeing to [2. above](https://github.com/cylc/cylc-admin/pull/189#issuecomment-2010574702) then?

Reply:
Happy to have a crack at that myself if you're busy on implementation?

Reply:
> I take it you are agreeing to https://github.com/cylc/cylc-admin/pull/189#issuecomment-2010574702 then?

No, I'm sticking with:

* Outputs are complete if the completion expression is satisfied.
* Tasks are complete if they have a final task status AND their outputs are complete.

Reminder, `TaskOutputs.is_complete` is an internal interface not exposed to users, this definition is not important.

Providing there aren't any technical qualms with this, I'm going to plug on with implementation as there's still quite a lot to do.

Reply:
But are you also dropping your proposed definition of *incomplete tasks*? If not, we're at an impasse. 

My bottom line is we can't redefine the terms complete and incomplete so that `"not complete" != "incomplete"`

Yes this isn't technical in the sense that we agree entirely on the logic, but it sure would be nice if the implementation didn't use inconsistent terms too.

Reply:
> eh? Tasks and TaskOutputs are two different things. I see no issue with what is written.

Your single-line change does not address my concerns without additional information.  As far as I can tell you are now saying that you won't call tasks complete if their outputs are complete, but you will still call them incomplete if their outputs are incomplete AND they are finished. That's a bit of a mess compared to either of my two ways to fix the inconsistent terminology.

You could stop spending time on this by simply agreeing - this is a review of a proposal and I've raised a valid objection to the terminology being used.

Reply:
As far as I can tell you are now saying that you won't call tasks complete if their outputs are complete, but you will still call them incomplete if their outputs are incomplete AND they are finished

Correct.

> You could stop spending time on this by simply agreeing

Harsh. Your turn to propose something, my turn to be nit picky.

Reply:
Sorry, didn't mean to be harsh - but I have proposed a solution (two of them in fact), in great detail, above. This is *only* a small question of consistent use of two common terms, it doesn't require a whole new proposal. Are you sure you've read and understood my "proposals"? (This sort of thing would be a lot easier to resolve if we didn't live in opposite timezones!)

Reply:
How about I merge this and immediately put up a tweak PR that redoes the use of "in/complete tasks" in the way that I propose?  (However, don't take that as full approval of this one yet :grin:)

Reply:
If you could pick an optional and raise it so I can review it in context.

Reply:
Ok, I'll clone your branch (rather than merge it) and put up an alternative.